src/Horarium/Handlers/ExecutorJob.cs

```diff
@@ -106,14 +106,16 @@ private async Task ExecuteJobRecurrent(JobMetadata jobMetadata)
 
                     _settings.Logger.Debug("jobMetadata excecuted");
 
-                    await _jobRepository.RemoveJob(jobMetadata.JobId);
+                    await _jobRepository.RescheduleRecurrentJob(jobMetadata.JobId,
+                        Utils.ParseAndGetNextOccurrence(jobMetadata.Cron), null);
 
                     _settings.Logger.Debug("jobMetadata saved success");
                 }
             }
             catch (Exception ex)
             {
-                await _jobRepository.FailedJob(jobMetadata.JobId, ex);
+                await _jobRepository.RescheduleRecurrentJob(jobMetadata.JobId,
+                    Utils.ParseAndGetNextOccurrence(jobMetadata.Cron), ex);
             }
             finally
```

Comment:
Do we still need this `finally` block here? Job will be resheduled now in any case (with exception or not)

Reply:
I think that was a mistake, I removed the finally block - https://github.com/Tinkoff/Horarium/pull/61/files